// ion-custom-scrollbar/fesm2015/ion-custom-scrollbar.js
import { __decorate } from 'tslib';
import { ElementRef, Input, Directive, NgModule } from '@angular/core';

let ScrollbarDirective = class ScrollbarDirective {
    constructor(elementRef) {
        this.elementRef = elementRef;
    }
    ngOnInit() {
        this.hostElement = this.elementRef.nativeElement;
        if (this.hostElement && this.hostElement.tagName && this.hostElement.tagName == 'ION-CONTENT') {
            let el = document.createElement('style');
            el.innerText = this.scrollbar || this.getCustomStyle();
            if (this.hostElement.shadowRoot) {
                this.hostElement.shadowRoot.appendChild(el);
            }
            else {
                this.hostElement.appendChild(el);
            }
        }
    }
    getCustomStyle() {
        if (this.allowAllScreens === true || this.allowAllScreens === 'true') {
            return `::-webkit-scrollbar {
          width: 12px;
        }
        ::-webkit-scrollbar-track {
          background: #fff;
        }
        ::-webkit-scrollbar-track:hover {
          background: #f7f7f7;
        }
        ::-webkit-scrollbar-thumb {
          background: #ccc;
        }
        ::-webkit-scrollbar-thumb:hover {
          background: #888
        }
        .inner-scroll {
          scrollbar-width: thin
        }`;
        }
        else {
            return `@media(pointer: fine) {
        ::-webkit-scrollbar {
          width: 12px;
        }
        ::-webkit-scrollbar-track {
          background: #fff;
        }
        ::-webkit-scrollbar-track:hover {
          background: #f7f7f7;
        }
        ::-webkit-scrollbar-thumb {
          background: #ccc;
        }
        ::-webkit-scrollbar-thumb:hover {
          background: #888
        }
        .inner-scroll {
          scrollbar-width: thin
        }
      }`;
        }
    }
};
ScrollbarDirective.ctorParameters = () => [
    { type: ElementRef }
];
__decorate([
    Input()
], ScrollbarDirective.prototype, "scrollbar", void 0);
__decorate([
    Input()
], ScrollbarDirective.prototype, "allowAllScreens", void 0);
ScrollbarDirective = __decorate([
    Directive({
        selector: '[scrollbar]'
    })
], ScrollbarDirective);

let IonCustomScrollbarModule = class IonCustomScrollbarModule {
};
IonCustomScrollbarModule = __decorate([
    NgModule({
        declarations: [ScrollbarDirective],
        imports: [],
        exports: [ScrollbarDirective]
    })
], IonCustomScrollbarModule);

/*
 * Public API Surface of ion-custom-scrollbar
 */

/**
 * Generated bundle index. Do not edit.
 */

export { IonCustomScrollbarModule, ScrollbarDirective };
