// ion-custom-scrollbar/fesm5/ion-custom-scrollbar.js
import { __decorate } from 'tslib';
import { ElementRef, Input, Directive, NgModule } from '@angular/core';

var ScrollbarDirective = /** @class */ (function () {
    function ScrollbarDirective(elementRef) {
        this.elementRef = elementRef;
    }
    ScrollbarDirective.prototype.ngOnInit = function () {
        this.hostElement = this.elementRef.nativeElement;
        if (this.hostElement && this.hostElement.tagName && this.hostElement.tagName == 'ION-CONTENT') {
            var el = document.createElement('style');
            el.innerText = this.scrollbar || this.getCustomStyle();
            if (this.hostElement.shadowRoot) {
                this.hostElement.shadowRoot.appendChild(el);
            }
            else {
                this.hostElement.appendChild(el);
            }
        }
    };
    ScrollbarDirective.prototype.getCustomStyle = function () {
        if (this.allowAllScreens === true || this.allowAllScreens === 'true') {
            return "::-webkit-scrollbar {\n          width: 12px;\n        }\n        ::-webkit-scrollbar-track {\n          background: #fff;\n        }\n        ::-webkit-scrollbar-track:hover {\n          background: #f7f7f7;\n        }\n        ::-webkit-scrollbar-thumb {\n          background: #ccc;\n        }\n        ::-webkit-scrollbar-thumb:hover {\n          background: #888\n        }\n        .inner-scroll {\n          scrollbar-width: thin\n        }";
        }
        else {
            return "@media(pointer: fine) {\n        ::-webkit-scrollbar {\n          width: 12px;\n        }\n        ::-webkit-scrollbar-track {\n          background: #fff;\n        }\n        ::-webkit-scrollbar-track:hover {\n          background: #f7f7f7;\n        }\n        ::-webkit-scrollbar-thumb {\n          background: #ccc;\n        }\n        ::-webkit-scrollbar-thumb:hover {\n          background: #888\n        }\n        .inner-scroll {\n          scrollbar-width: thin\n        }\n      }";
        }
    };
    ScrollbarDirective.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    __decorate([
        Input()
    ], ScrollbarDirective.prototype, "scrollbar", void 0);
    __decorate([
        Input()
    ], ScrollbarDirective.prototype, "allowAllScreens", void 0);
    ScrollbarDirective = __decorate([
        Directive({
            selector: '[scrollbar]'
        })
    ], ScrollbarDirective);
    return ScrollbarDirective;
}());

var IonCustomScrollbarModule = /** @class */ (function () {
    function IonCustomScrollbarModule() {
    }
    IonCustomScrollbarModule = __decorate([
        NgModule({
            declarations: [ScrollbarDirective],
            imports: [],
            exports: [ScrollbarDirective]
        })
    ], IonCustomScrollbarModule);
    return IonCustomScrollbarModule;
}());

/*
 * Public API Surface of ion-custom-scrollbar
 */

/**
 * Generated bundle index. Do not edit.
 */

export { IonCustomScrollbarModule, ScrollbarDirective };
